packages/remix-react/components.tsx

```diff
@@ -859,10 +1056,107 @@ window.__remixRouteModules = {${matches
         />
       ))}
       {isHydrated ? null : initialScripts}
+      <DeferredHydrationScripts />
     </>
   );
 }
 
+function DeferredHydrationScripts() {
+  let { routeData, deferredLoaderData } = useRemixEntryContext();
+
+  let scripts = React.useMemo<null | React.ReactNode[]>(() => {
+    if (!deferredLoaderData) {
+      return null;
+    }
+
+    let scripts: React.ReactNode[] = [];
+    for (let [routeId, deferredKeys] of Object.entries(
+      deferredLoaderData || {}
+    )) {
+      if (!routeData[routeId] || typeof routeData[routeId] !== "object") {
```

Comment:
Needs a promise check.

Reply:
Not actually needed after double checking.